lib/apiservers/engine/backends/cache/container_cache.go

```diff
@@ -129,3 +131,51 @@ func (cc *CCache) GetContainerFromExec(eid string) *container.VicContainer {
 	}
 	return nil
 }
+
+// here we assume that the newName is already reserved by Reservename
+// so no need to check the existence of a container with the new name
+func (cc *CCache) UpdateContainerName(oldName, newName string) error {
+	cc.m.Lock()
+	defer cc.m.Unlock()
+
+	container := cc.getContainer(oldName)
+	if container == nil {
+		return derr.NewRequestNotFoundError(fmt.Errorf("no such container: %s", oldName))
+	}
+
+	delete(cc.containersByName, container.Name)
+
+	container.Name = newName
+	cc.containersByName[newName] = container
+	cc.containersByID[container.ContainerID] = container
+
+	return nil
+}
+
+// reserve the name for container
```

Comment:
Please expand upon the function comment - we can say
```
// ReserveName is used during a container rename operation to prevent concurrent
// container create operations from grabbing the new name.
```